controller/cmd/service-mirror/cluster_watcher.go

```diff
@@ -910,3 +962,61 @@ func (rcsw *RemoteClusterServiceWatcher) Stop(cleanupState bool) {
 	}
 	rcsw.eventsQueue.ShutDown()
 }
+
+func (rcsw *RemoteClusterServiceWatcher) extractGatewaySpec(gateway *corev1.Service) (*GatewaySpec, error) {
+	if len(gateway.Status.LoadBalancer.Ingress) == 0 {
+		return nil, errors.New("expected gateway to have at lest 1 external Ip address but it has none")
+	}
+
+	var foundPort = false
+	var port uint32
+	for _, p := range gateway.Spec.Ports {
+		if p.Name == consts.GatewayPortName {
+			foundPort = true
+			port = uint32(p.Port)
+			break
+		}
+	}
+
+	if !foundPort {
+		return nil, fmt.Errorf("cannot find  port named %s on gateway", consts.GatewayPortName)
+	}
+
+	var gatewayEndpoints []corev1.EndpointAddress
+	for _, ingress := range gateway.Status.LoadBalancer.Ingress {
+		gatewayEndpoints = append(gatewayEndpoints, corev1.EndpointAddress{
+			IP:       ingress.IP,
+			Hostname: ingress.Hostname,
+		})
+	}
+
+	gatewayIdentity := gateway.Annotations[consts.GatewayIdentity]
+
+	var probeCfg *ProbeConfig
+
+	probePath := gateway.Annotations[consts.GatewayProbePath]
```

Comment:
should we log if any of these errs are not nil?  if we move all the parsing of the ProbeConfig into a separate method that returns `(*ProbeConfig, error)` we can be a bit more idiomatic about error handling and avoid nesting if statements so deeply.